src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -32,7 +35,7 @@ public class CodeGenIL : IDisposable
         /// </summary>
         private Dictionary<int, Type> astTypeInfoMap = new Dictionary<int, Type>();
         private StreamWriter writer;
-        private Dictionary<int, IEnumerable<MethodBase>> methodCache = new Dictionary<int, IEnumerable<MethodBase>>();
+        private Dictionary<int, IEnumerable<ProtoCore.CLRFunctionEndPoint>> methodCache = new Dictionary<int, IEnumerable<ProtoCore.CLRFunctionEndPoint>>();
```

Comment:
I need to stare at this for a while - but let's say that we want to emit some IL that directly calls a function, without wrapping the return types in CLRStackValue and without using replication logic at runtime - does this change stop us from doing that?

Reply:
Not really. It would just mean to call other APIs on the  ProtoCore.CLRFunctionEndPoint class instead of using MethodBase